kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/impl/ContainerOrchestrationServiceImpl.java

```diff
@@ -684,45 +683,45 @@ private HostConfig containerRuntimeConfigurationHandler(ContainerConfiguration c
-            HostConfig commandBuilder) {
+    private HostConfig containerPortManagementHandler(ContainerConfiguration containerConfiguration,
+            HostConfig hostConfig, CreateContainerCmd commandBuilder) {
 
-        if (containerDescription.getContainerPorts() != null && !containerDescription.getContainerPorts().isEmpty()) {
-            List<ExposedPort> exposedPortsList = new LinkedList<>();
-            Ports portbindings = new Ports();
+        List<ExposedPort> exposedPorts = new LinkedList<>();
+        Ports portbindings = new Ports();
 
-            for (org.eclipse.kura.container.orchestration.ContainerPort port : containerDescription
-                    .getContainerPorts()) {
+        if (containerConfiguration.getContainerPorts() != null
+                && !containerConfiguration.getContainerPorts().isEmpty()) {
 
-                InternetProtocol ipPro;
+            List<org.eclipse.kura.container.orchestration.ContainerPort> containerPorts = containerConfiguration
+                    .getContainerPorts();
 
-                switch (port.getInternetProtocol()) {
-                case UDP:
-                    ipPro = InternetProtocol.UDP;
-                    break;
-                case SCTP:
-                    ipPro = InternetProtocol.SCTP;
-                    break;
-                default:
-                    ipPro = InternetProtocol.TCP;
-                    break;
-                }
+            for (org.eclipse.kura.container.orchestration.ContainerPort port : containerPorts) {
 
-                ExposedPort tempExposedPort = new ExposedPort(port.getInternalPort(), ipPro);
-                exposedPortsList.add(tempExposedPort);
-                portbindings.bind(tempExposedPort, Binding.bindPort(port.getExternalPort()));
-            }
+                InternetProtocol ipProtocol = InternetProtocol.TCP;
 
-            commandBuilder.withPortBindings(portbindings);
+                if (port.getInternetProtocol() != null) {
+                    try {
+                        ipProtocol = InternetProtocol.parse(port.getInternetProtocol().toString());
+                    } catch (IllegalArgumentException e) {
+                        logger.warn("Invalid internet protocol: {}. Using TCP.", port.getInternetProtocol());
+                    }
+                }
 
-            this.exposedPorts = exposedPortsList;
+                ExposedPort exposedPort = new ExposedPort(port.getInternalPort(), ipProtocol);
+                exposedPorts.add(exposedPort);
+                portbindings.bind(exposedPort, Binding.bindPort(port.getExternalPort()));
+            }
 
-        } else {
-            logger.error("portsExternal and portsInternal must be int[] of the same size or they do not exist: {}",
-                    containerDescription.getContainerName());
+            if (exposedPorts.size() != portbindings.getBindings().size()) {
```

Comment:
Is it possible for this to be false?

Reply:
Maybe if there are multiple entries for the same port in received configuration https://github.com/docker-java/docker-java/blob/b29527510d9f7c16b30b138ae4a2f6703d88bb01/docker-java-api/src/main/java/com/github/dockerjava/api/model/Ports.java#L51

Reply:
The description of the metatype says that the size must be the same:

https://github.com/eclipse/kura/blob/6629ddf759d88af03b1da88bc27a3fc82253ce38/kura/org.eclipse.kura.container.provider/OSGI-INF/metatype/org.eclipse.kura.container.provider.ContainerInstance.xml#L91-L99

Reply:
That part is related to the ContainerInstance configurable component, which is a client of the ContainerOrchestrationService API, so it shouldn't be relevant here, since ContainerOrchestrationServiceImpl only considers the result of ContainerConfiguration.getContainerPorts().

Anyway if the user specifies a list of Internal Ports with a different length than External Ports in ContanierInstance, the configuration parsing logic of that component will probably fail https://github.com/eclipse/kura/blob/6629ddf759d88af03b1da88bc27a3fc82253ce38/kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstanceOptions.java#L371

The condition here could probably evaluate to false if the user specifies multiple container ports with the same internal/external port, in the future we should investigate if that configuration is valid and what happens in that case.